src/domains/portfolio/hooks/use-portfolio.ts

```diff
@@ -121,7 +121,7 @@ export const usePortfolio = ({ profile }: { profile: Contracts.IProfile }) => {
 	const allAddresses = addresses.all();
 
 	useEffect(() => {
-		if (selectedAddresses.length === 0) {
+		if (selectedAddresses.length === 0 && allAddresses.length > 0 && !isEqual(selectedAddresses, allAddresses)) {
```

Comment:
```suggestion
		if (selectedAddresses.length === 0 && !isEqual(selectedAddresses, allAddresses)) {
```
this would have the same effect right?